File: nbcamp-java-spring-crud-example/src/main/java/com/navi/nbcampjavaspringcrudexample/api/controller/MemberController.java
package com.navi.nbcampjavaspringcrudexample.api.controller;

import com.navi.nbcampjavaspringcrudexample.api.model.MemberRegisterRequest;
import com.navi.nbcampjavaspringcrudexample.database.entity.Member;
import com.navi.nbcampjavaspringcrudexample.database.repository.MemberRepository;
import java.time.LocalDateTime;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class MemberController {

    private final MemberRepository memberRepository;


    @PostMapping("/members")
    public Member registerMember(@RequestBody MemberRegisterRequest request) {
        Member member = new Member(request.getName(), LocalDateTime.now(), LocalDateTime.now());

        return memberRepository.save(member);
    }

    @GetMapping("/members/{id}")
    public Member getMemberById(@PathVariable Long id) {
        return memberRepository.findById(id)
            .orElseThrow(() -> new IllegalArgumentException("멤버가 없습니다!!!"));
    }

}

File: nbcamp-kotlin-crud-example/src/main/kotlin/com/navi/nbcampkotlincrudexample/api/controller/MemberController.kt
package com.navi.nbcampkotlincrudpractice.api.controller

import com.navi.nbcampkotlincrudpractice.api.model.member.RegisterMemberRequest
import com.navi.nbcampkotlincrudpractice.api.service.MemberService
import com.navi.nbcampkotlincrudpractice.database.entity.Member
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
@RequestMapping("/api/members")
class MemberController(
    private val memberService: MemberService
) {
    @GetMapping("/{id}")
    fun findById(
        @PathVariable("id") id: Long,
    ): Member = memberService.findById(id)

    @GetMapping
    fun findAll(): List<Member> = memberService.findAll()

    @PostMapping("/register")
    fun register(
        @RequestBody request: RegisterMemberRequest,
    ): Member =
        memberService.register(
            request.username,
            request.password,
            request.email,
            LocalDateTime.now(),
        )
}
